docs/reference/index.md

```diff
@@ -66,20 +65,34 @@ Language clients are **forward compatible**: each client version works with equi
 
 Compatibility does not imply feature parity. New {{es}} features are supported only in equivalent client versions. For example, an 8.12 client fully supports {{es}} 8.12 features and works with 8.13 without breaking, but it does not support new {{es}} 8.13 features. An 8.13 client fully supports {{es}} 8.13 features.
 
-{{es}} language clients are also **backward compatible** across minor versions, with default distributions and without guarantees. 
+{{es}} language clients are also **backward compatible** across minor versions, with default distributions and without guarantees.
 
-### Major version upgrades
+## Upgrade to a new major version
 
 :::{important}
 To upgrade to a new major version, first [upgrade {{es}}](docs-content://deploy-manage/upgrade.md), then upgrade the Python client.
 :::
 
-As of version 8.0, {{es}} offers a [compatibility mode](elasticsearch://reference/elasticsearch/rest-apis/compatibility.md) for smoother upgrades. In compatibility mode, you can upgrade your {{es}} cluster to the next major version while continuing to use your existing client during the transition. 
+As of version 8.0, {{es}} offers a [REST API compatibility mode](elasticsearch://reference/elasticsearch/rest-apis/compatibility.md) for smoother upgrades. In compatibility mode, you can upgrade your {{es}} cluster to the next major version while continuing to use your existing client during the transition.
+
+### How compatibility mode works
+
+In the Python client, compatibility mode is **always enabled**. You don't need to configure it manually or set an environment variable. The client automatically sends compatibility headers with API requests so that it can communicate with the next major version of {{es}}.
 
-For example, if you're upgrading {{es}} from 8.x to 9.x, you can continue to use the 8.x Python client during and after the {{es}} server upgrade, with the exception of [breaking changes](../release-notes/breaking-changes.md).
+In compatibility mode, your Python client applications can continue running during an upgrade without restart or redeployment. For example, during a rolling upgrade of your {{es}} cluster from 8.x to 9.x, your 8.x Python client continues working throughout the process without any changes or intervention.
```

Comment:
This seems a bit less clear than the original text. Here we are saying "without changes or intervention", but right below we go on to explain that this isn't always true because of breaking changes. We also repeat "working without intervention" in line 92.

I think it would be more accurate to say something like "generally without any changes or interventionm with some exceptions mentioned below" or something along those lines.